### PR TITLE
End user discussion summary for March

### PR DESCRIPTION
**Pull request by reese-lee**
This is the summary of the end user discussions across all regions for the month of March. 

---

Preview: https://deploy-preview-2560--opentelemetry.netlify.app/blog/2023/end-user-discussions-03/

### CODE REVIEW

File: content/en/blog/2023/end-user-discussions-03.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+[http check receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/healthcheckextension/README.md).
+
+#### 3 - Collector distributions
+
+**Q:** Should I use a vendor distribution versus the community collector
+distribution?
+
+**A:** Each vendor distribution will come with customizations, whereas the
+community Collector distribution will include everything: receivers and
+exporters. If you need the flexibility, then you should use the OTel Collector
```
Comment (avillela) [inline content/en/blog/2023/end-user-discussions-03.md:63]:
```suggestion
exporters. If you need the flexibility, then you should use the OTel community Collector
```

File: content/en/blog/2023/end-user-discussions-03.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+distro.
+
+#### 4 - Rate limiting on receivers
+
+**Q:** Each vendor distribution will come with customizations, whereas the
```
Comment (avillela) [inline content/en/blog/2023/end-user-discussions-03.md:68]: I think this is a duplication of the previous question.

File: content/en/blog/2023/end-user-discussions-03.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+point. One approach might be to add additional headers when you configure your
+SDKs that contain the additional information, which would help with load
+balancing.
+
+#### 5 - Connectors
+
+**Q:** What are connectors?
+
+**A:**
+[Read about connectors here](https://o11y.news/2023-03-13/#opentelemetry-connectors).
```
Comment (avillela) [inline content/en/blog/2023/end-user-discussions-03.md:90]: Would it be worth providing a high-level explanation before posting the link?

File: content/en/blog/2023/end-user-discussions-03.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+
+#### 6 - Definitions of upstream, downstream, and distro
+
+**Q:** What is upstream? Downstream? Distro?
+
+**A:** Check out
+[this documentation](https://opentelemetry.io/ecosystem/vendors/) and this
```
Comment (avillela) [inline content/en/blog/2023/end-user-discussions-03.md:97]: Again, would it be helpful to include high-level definitions here before referring readers to the links?

File: content/en/blog/2023/end-user-discussions-03.md [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+than the technical challenges. The key is to not propose one big change; the
+journey of moving from a vendor-based solution to OpenTelemetry should be a
+step-by-step process, rather than going into a full transformation.
+
+Additional suggestions:
+
+- Start with dev or testing environments first to build trust in the software
+- Choose a stack where OTel is more robust, such as Java and Node.js
+- For countering developer resistance, using auto-instrumentation modules to
+- start with is a good step
```
Comment (avillela) [inline content/en/blog/2023/end-user-discussions-03.md:153]:
```suggestion
start with is a good step
```

File: content/en/blog/2023/end-user-discussions-03.md [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+user noticed a 20% reduction in CPU usage when they moved from a vendor-specific
+agent to the host metrics receiver for infrastructure monitoring.
+
+#### 3 - Auto-instrumentation
+
+**Q:** Is there a way to automatically create spans without code changes?
+
+**A:** It depends on the use cases:
+
+- Auto instrumentation options are maturing in OTel; for example, the Java JAR
```
Comment (avillela) [inline content/en/blog/2023/end-user-discussions-03.md:184]: Is it worth mentioning auto-instrumentation also available for Python, NodeJS, and .NET?

File: content/en/blog/2023/end-user-discussions-03.md [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+- agent takes care of instrumenting
+  [most libraries](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks)
+  that are used by applications.
+- If you’re using Kubernetes, they can use the
+  [OTel operator](https://github.com/open-telemetry/opentelemetry-operator),
+- which takes care of instrumentations for applications deployed on k8s.
```
Comment (avillela) [inline content/en/blog/2023/end-user-discussions-03.md:190]: More specifically, the OTel Operator also supports injecting and configuring [auto-instrumentation](https://opentelemetry.io/docs/concepts/instrumenting/#automatic-instrumentation) libraries for services written in [Python](https://opentelemetry.io/docs/instrumentation/python/automatic/), [Java](https://opentelemetry.io/docs/instrumentation/java/automatic/), [.NET](https://opentelemetry.io/docs/instrumentation/net/automatic/), and [NodeJS](https://opentelemetry.io/docs/instrumentation/js/libraries/#node-autoinstrumentation-package).

File: content/en/blog/2023/end-user-discussions-03.md [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+- If you’re using AWS lambda, you should check out the
+  [OTel Lambda extension](https://github.com/open-telemetry/opentelemetry-lambda).
+
+#### 4 - Leveraging telemetry from OTel
+
+**Q:** Has there been work toward tele-command standards to leverage the
```
Comment (avillela) [inline content/en/blog/2023/end-user-discussions-03.md:196]: Elaborate on what tele-command is, for those who aren't aware of it

File: content/en/blog/2023/end-user-discussions-03.md [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+[Observability Day](https://events.linuxfoundation.org/kubecon-cloudnativecon-europe/co-located-events/observability-day/).
+Join the discussion at
+[#telemetry-analysis](https://cloud-native.slack.com/archives/C04LXHPDW6M) in
+CNCF’s Slack instance.
+
+#### 2 - Documentation and searches
+
+**Q:** Where do you go to find documentation and answers to your questions?
+
+**A:** It would be helpful to gather feedback from end users – do you go to the
```
Comment (avillela) [inline content/en/blog/2023/end-user-discussions-03.md:238]: This reads more like a question than an answer, so I'm finding it a tad confusing. 😵‍💫

File: content/en/blog/2023/end-user-discussions-03.md (hunk 2 above)
Comment (reese-lee) [reply content/en/blog/2023/end-user-discussions-03.md:68]: Ooo bad copy-paste job on my part (I started the post in a Google doc). Thanks for catching this!

File: content/en/blog/2023/end-user-discussions-03.md [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+**A:** The terms "upstream" and "downstream" refer to how services or components
+in a system are connected to each other. Check out
+[this article](https://reflectoring.io/upstream-downstream/) for more
+information as it applies to different situations in software.
+
+The term "distro" is short for distribution. Check out
+[this documentation](https://opentelemetry.io/ecosystem/vendors/) to see which
+vendors provide distros.
+
+### SAMPLING
```
Comment (chalin) [inline content/en/blog/2023/end-user-discussions-03.md:105]: I'd suggest regular "sentence case" for all headings (let's avoid ALL CAPS here and elsewhere if we can).

 ```suggestion
### Sampling
```

File: content/en/blog/2023/end-user-discussions-03.md [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+## Join us!
+
+If you have a story to share about how you use OpenTelemetry at your
+organization, we’d love to hear from you! Ways to share:
+
+- Join the [#otel-endusers channel](/community/end-user/slack-channel/) on the
+  [CNCF Community Slack](https://communityinviter.com/apps/cloud-native/cncf)
+- Join our monthly
+  [End-User Discussion Group calls](/community/end-user/discussion-group/)
+- [Join our OTel in Practice sessions](/community/end-user/otel-in-practice/)
```
Comment (chalin) [inline content/en/blog/2023/end-user-discussions-03.md:281]: For consistency with the other list items:

```suggestion
- Join our [OTel in Practice](/community/end-user/otel-in-practice/) sessions
```

File: content/en/blog/2023/end-user-discussions-03.md [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+**Q:** What is upstream? Downstream? Distro?
+
+**A:** The terms "upstream" and "downstream" refer to how services or components
+in a system are connected to each other. Check out
+[this article](https://reflectoring.io/upstream-downstream/) for more
+information as it applies to different situations in software.
+
+The term "distro" is short for distribution. Check out
+[this documentation](https://opentelemetry.io/ecosystem/vendors/) to see which
+vendors provide distros.
```
Comment (chalin) [inline content/en/blog/2023/end-user-discussions-03.md:103]:
```suggestion
The term "distro" is short for distribution. For a list of vendors that provide distros,
see [Vendors](/ecosystem/vendors/).
```

File: content/en/blog/2023/end-user-discussions-03.md [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+you have a data ingest or storage cost? If you have ingest cost, you’ll want to
+sample before the data gets ingested; if it’s storage cost, you’ll have to
+delete a lot of the data, so it depends on the tradeoffs.
+
+One thing to consider is that you can use tail sampling on attributes, such as
+if there’s an error on a span, but it does require more memory. Suggested
+further exploration:
+
+- [Column data store for OpenTelemetry](https://github.com/open-telemetry/oteps/pull/171)
+- [OpAMP](https://opentelemetry.io/blog/2022/opamp/)
```
Comment (chalin) [inline content/en/blog/2023/end-user-discussions-03.md:128]: Drop `https://opentelemetry.io` from every link in this page that refers to a website local path:

```suggestion
- [OpAMP](/blog/2022/opamp/)
```

**Comment by reese-lee** [comment]
@chalin Do I run that in the top opentelemetry.io directory or somewhere else? I'm getting a `sh: run-s: command not found` when running in the top-level directory.

**Comment by chalin** [comment]
@reese-lee: yes, top-level directory. Hmm. Did you run `npm install` first (to get all required packages)?

**Comment by chalin** [comment]
If that doesn't work out, let me know and I'll refresh the refcache file for you.

File: content/en/blog/2023/end-user-discussions-03.md [hunk 14]
```diff
@@ -0,0 +1,287 @@
...
+if there’s an error on a span, but it does require more memory. Suggested
+further exploration:
+
+- [Column data store for OpenTelemetry](https://github.com/open-telemetry/oteps/pull/171)
+- [OpAMP](/blog/2022/opamp/)
+- Your backend vendor’s tail-based sampling strategies
+- [Paper by Uber](https://www.uber.com/en-IN/blog/crisp-critical-path-analysis-for-microservice-architectures/)
+- [Tail sampling processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/tailsamplingprocessor/README.md)
+
+### ADOPTION, MIGRATION, AND IMPLEMENTATION
```
Comment (chalin) [inline content/en/blog/2023/end-user-discussions-03.md:132]:
```suggestion
### Adoption, Migration, and Implementation
```

File: content/en/blog/2023/end-user-discussions-03.md [hunk 15]
```diff
@@ -0,0 +1,287 @@
...
+[OpAMP](https://opentelemetry.io/blog/2022/opamp/).
+
+#### 5 - Message brokers
+
+**Q:** What are some use cases for message brokers?
+
+**A:** IoT use cases (car manufacturer). There is also ongoing work for semantic
+conventions support for messages.
+
+### UPDATES AND COMMUNICATIONS
```
Comment (chalin) [inline content/en/blog/2023/end-user-discussions-03.md:222]:
```suggestion
### Updates and Communications
```

File: content/en/blog/2023/end-user-discussions-03.md (hunk 13 above)
Comment (chalin) [reply content/en/blog/2023/end-user-discussions-03.md]: > Drop `https://opentelemetry.io` from _**every link**_ in this page that refers to a website local path

Marking as unresolved: there are other links that need to have the `https://opentelemetry.io` dropped. If you'd rather I make that change, let me know.

Comment (reese-lee) [reply content/en/blog/2023/end-user-discussions-03.md]: Oh whoops! I'll go through the links.

Comment (reese-lee) [reply content/en/blog/2023/end-user-discussions-03.md]: Ok I think I got them all.

**Comment by reese-lee** [comment]
@chalin Can you refresh the refcache file? I'm running into other issues with `npm install`.

**Comment by chalin** [comment]
> @chalin Can you refresh the refcache file? I'm running into other issues with `npm install`.

Done ✨ 
Rebased, updated the refcache and reran Prettier.

**Review by cartermp — APPROVED** [review]
Love this! I am ready to have these merged whenever.

**Review by chalin — APPROVED** [review]

**Pull request merged**